File: liteadapter/src/main/java/me/yuu/liteadapter/core/DataOperator.java
package me.yuu.liteadapter.core;

import java.util.List;

/**
 * @author yu.
 * @date 2018/1/12
 */
public interface DataOperator<D> {

    D getItem(int position);

    void addData(D item);

    void addData(int position, D item);

    void addAll(List<D> items);

    void addAll(int position, List<D> items);

    void remove(int position);

    void modify(int position, D newData);

    void modify(int position, Action<D> action);

    /**
     * 设置新的数据集合，自动应用DiffUtil进行差量更新
     */
    void updateData(List<D> items);

    void clear();

    interface Action<T> {
        void doAction(T data);
    }
}

File: liteadapter/src/main/java/me/yuu/liteadapter/core/DataOperator.kt
package me.yuu.liteadapter.core

/**
 * @author yu.
 * @date 2018/1/12
 */
interface DataOperator<D> {
    fun getItem(position: Int): D

    fun addData(item: D)
    fun addData(position: Int, item: D)
    fun addAll(items: List<D>)
    fun addAll(position: Int, items: List<D>)
    fun remove(position: Int)
    fun modify(position: Int, newData: D)
    fun modify(position: Int, action: (D) -> Unit)

    /**
     * 设置新的数据集合，自动应用DiffUtil进行差量更新
     */
    fun updateData(items: List<D>)

    fun clear()
}